binding/flutter/lib/porcupine.dart

```diff
@@ -92,121 +106,116 @@ class Porcupine {
   /// Thows a `PvError` if not initialized correctly
   ///
   /// returns an instance of the wake word engine
-  static Future<Porcupine> fromKeywordPaths(List<String?>? keywordPaths,
-          {String? modelPath, List<double>? sensitivities}) =>
-      Porcupine._create(
-          keywordPaths: keywordPaths,
-          modelPath: modelPath,
-          sensitivities: sensitivities);
+  static Future<Porcupine> fromKeywordPaths(
+      String accessKey, List<String?>? keywordPaths,
+      {String? modelPath, List<double>? sensitivities}) async {
+    await _init();
+
+    return Porcupine._create(
+        accessKey: accessKey,
+        keywordPaths: keywordPaths,
+        modelPath: modelPath,
+        sensitivities: sensitivities);
+  }
 
   /// private static creator
   static Future<Porcupine> _create(
-      {List<String?>? keywords,
+      {String? accessKey,
       List<String?>? keywordPaths,
       String? modelPath,
       List<double>? sensitivities}) async {
-    if (!_resourcesExtracted) {
-      await _extractPorcupineResources();
-      _resourcesExtracted = true;
+    if (accessKey == null || accessKey == "") {
+      throw PorcupineInvalidArgumentException("'accessKey' must be set.");
     }
 
     if (modelPath == null || modelPath == "") {
       modelPath = _defaultModelPath;
     }
 
-    if (!await File(modelPath!).exists()) {
+    if (!(await File(modelPath!).exists())) {
       try {
         modelPath = await _extractResource(modelPath);
-      } catch (_) {
-        throw new PvArgumentError(
-            "Could not find model file at path '$modelPath'. If this is a packaged asset, ensure you have added it to your pubspec.yaml file.");
-      }
-    }
-
-    if (keywords != null && keywordPaths != null) {
-      throw new PvArgumentError(
-          "Both 'keywords' and 'keywordPaths' were set. Only one of the two arguments may be set for initialization.");
+      } catch (e) {/* don't do anything on fail */}
```

Comment:
hmmm...not sure about this. I know we're trying to allow auto-extraction in Flutter and native, but there must be a more explicit way of checking this so that it doesn't just do a blind catch.